orderer/common/follower/block_puller.go

```diff
@@ -142,27 +142,27 @@ func (creator *BlockPullerCreator) BlockFetcher(configBlock *common.Block, stopC
 		CensorshipSuspicionThreshold: time.Duration((int64(creator.clusterConfig.ReplicationPullTimeout) * shuffleTimeoutPercentage / 100)),
 		PeriodicalShuffleInterval:    shuffleTimeoutMultiplier * creator.clusterConfig.ReplicationPullTimeout,
 		MaxRetries:                   uint64(creator.clusterConfig.ReplicationMaxRetries),
-		MaxByzantineNodes:            (len(endpoints) - 1) / 3,
 	}
 
-	blockVerifierFunc := func(header *common.BlockHeader, metadata *common.BlockMetadata) error {
-		return creator.VerifyBlockSequence([]*common.Block{
-			{
-				Header:   header,
-				Metadata: metadata,
-			},
-		}, creator.channelID)
+	// To tolerate byzantine behaviour of `f` faulty nodes, we need atleast of `3f + 1` nodes.
+	// check for bft enable and update `MaxByzantineNodes`
+	// accordingly.
+	bftEnabled := cluster.BFTEnabledInConfig(configBlock, creator.bccsp)
+	if bftEnabled {
+		fc.MaxByzantineNodes = int((len(endpoints) - 1) / 3)
```

Comment:
The number of endpoints should not govern the number of byzantine nodes, but rather the number of consenters in the channel config: `[]*common.Consenter` from `bundle.OrdererConfig().Consenters()`

Reply:
Can we use the same formula: fc.MaxByzantineNodes = int((len(consenters) - 1) / 3)?

Reply:
Use [this](https://github.com/SmartBFT-Go/consensus/blob/57ed2ce2c73b0e2f4fd88d941c5afe7131ea7aae/internal/bft/util.go#L159-L163) function.

Reply:
Yes